### ISSUE TITLE
Strange behaviour with spring-boot (or jackson-databind:2.9.9)

### ISSUE THREAD

**Issue by fathzer** [bohnman/squiggly#67]
Here is a simple example to reproduce the problem:

```
import java.util.Arrays;
import java.util.Collection;
import java.util.TreeSet;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.github.bohnman.squiggly.Squiggly;
import com.github.bohnman.squiggly.util.SquigglyUtils;

public class SquigglyTest {
	public static void main(String[] args) {
		Object object = new ListResponse(Arrays.asList(new Item[] {new Item("id")}));
		
		ObjectMapper mapper = Squiggly.init(new ObjectMapper(), "items[**]");
		System.out.println(SquigglyUtils.stringify(mapper, object));

		mapper = Squiggly.init(new ObjectMapper(), "items");
		// Should return set and id ... but returns only set when spring-boot-parent is in the pom! 
		System.out.println(SquigglyUtils.stringify(mapper, object));
	}

	public static class ListResponse {
		private Collection<Item> items;

		public ListResponse(Collection<Item> items) {
			super();
			this.items = items;
		}

		public Collection<Item> getItems() {
			return items;
		}
	}
		
	public static class Item {
		private Collection<String> set;
		private String id;
		
		public Item(String id) {
			// Strange if we create a List instead of a Set, the problem vanishes
			this.set = new TreeSet<>();
			this.id = id;
		}

		public Collection<String> getSet() {
			return set;
		}

		public String getId() {
			return id;
		}
	}
}
```
The non filtered object is: {"items":[{"set":[],"id":"id"}]}
The program filters it with the "items[**]" and "items" filters.
If the only dependency in the pom is squiggly library (version 1.3.18), anything is fine and the output is the one expected:

> {"items":[{"set":[],"id":"id"}]}
> {"items":[{"set":[],"id":"id"}]}

Now, if we add the spring-boot-parent in the pom
```
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>2.1.7.RELEASE</version>
	</parent>
```
The output become:

> {"items":[{"set":[],"id":"id"}]}
> {"items":[{"set":[]}]}

The "items" filter doesn't work anymore.

After digging a little I found the problem is trigerred because the collection in the Item class is a TreeSet (with an ArrayList, the problem vanishes).
The important difference between with and without springboot parent is the version of jackson databind (2.9.9 in spring-boot dependency management instead of 2.6.4 declared in squiggly pom).
Removing the spring-boot parent and adding an explicit dependency to jackson-databind:2.9.9 makes the problem also occur.

**Comment by bohnman** [comment]
@fathzer Sorry for the late reply.  This turned out to be a bug in Jackson. I have submitted the following issue in the project:

https://github.com/FasterXML/jackson-databind/issues/2475

**Comment by bohnman** [comment]
Looks like this issue will be fixed starting in Jackson version 2.10.1

**Comment by fathzer** [comment]
Hi,
Jackson 2.10.1 was released ... and it fixed it.
Thanks a lot :-)
